src/Containers/Microsoft.NET.Build.Containers/Registry/Registry.cs

```diff
@@ -7,9 +7,49 @@
 using System.Diagnostics;
 using System.Net.Http.Json;
 using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
 
 namespace Microsoft.NET.Build.Containers;
 
+internal interface IManifestPicker {
+    public PlatformSpecificManifest? PickBestManifestForRid(IReadOnlyDictionary<string, PlatformSpecificManifest> manifestList, string runtimeIdentifier);
+}
+
+internal sealed class RidGraphManifestPicker : IManifestPicker
+{
+    private readonly RuntimeGraph _runtimeGraph;
+
+    public RidGraphManifestPicker(string runtimeIdentifierGraphPath)
+    {
+        _runtimeGraph = GetRuntimeGraphForDotNet(runtimeIdentifierGraphPath);
+    }
+    public PlatformSpecificManifest? PickBestManifestForRid(IReadOnlyDictionary<string, PlatformSpecificManifest> ridManifestDict, string runtimeIdentifier)
+    {
+        var bestManifestRid = GetBestMatchingRid(_runtimeGraph, runtimeIdentifier, ridManifestDict.Keys);
+        if (bestManifestRid is null) {
+            return null;
+        }
+        return ridManifestDict[bestManifestRid];
+    }
+
+    private static string? GetBestMatchingRid(RuntimeGraph runtimeGraph, string runtimeIdentifier, IEnumerable<string> availableRuntimeIdentifiers)
```

Comment:
nit: Why is this method `static`? :) It's invoked form instance method and first parametr represents instance data.

Reply:
In general I prefer static functions - that way a reader knows that no other state/inputs other than the function parameters is required to understand the function. The fancy term is 'locality of reference' and it's a concept that all of us functional programming types are prone to doing :)